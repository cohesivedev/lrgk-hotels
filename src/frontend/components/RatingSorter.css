.RatingSorter {
    border:  1px solid black;
    padding: 1em;
}

.RatingSorter select {
    margin-left: 1em;
}
