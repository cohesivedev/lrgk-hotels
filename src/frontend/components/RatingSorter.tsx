import * as React from 'react';
import {ChangeEventHandler} from 'react';

import './RatingSorter.css';

interface IRatingSorterProps {
    isAscending: boolean;
    onChange: ChangeEventHandler;
}

const RatingSorter: React.FC<IRatingSorterProps> = ({isAscending, onChange}) => (
    <div className="RatingSorter">
        <label htmlFor="rating-sorter">
            Show me hotels by star rating
            <select id="rating-sorter" onChange={onChange} value={isAscending ? 'ascending' : 'descending'}>
                <option value="ascending">Low to High (Ascending)</option>
                <option value="descending">High to Low (Descending)</option>
            </select>
        </label>
    </div>
);

export default RatingSorter;