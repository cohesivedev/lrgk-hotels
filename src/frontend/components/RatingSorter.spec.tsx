import * as React from 'react';
import RatingSorter from './RatingSorter';
import {renderToString} from 'react-dom/server';

describe('components/RatingSorter', () => {
    it('should match snapshot for ascending',
        () => expect(
            renderToString(
                <RatingSorter
                    isAscending
                    onChange={new Function() as any}
                />
            )
        ).toMatchSnapshot()
    );

    it('should match snapshot for descending',
        () => expect(
            renderToString(
                <RatingSorter
                    isAscending={false}
                    onChange={new Function() as any}
                />
            )
        ).toMatchSnapshot()
    );
});
