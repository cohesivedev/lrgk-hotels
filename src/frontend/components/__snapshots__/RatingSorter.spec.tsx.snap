// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RatingSorter should match snapshot for ascending 1`] = `

<div class="RatingSorter"
     data-reactroot
>
  <label for="rating-sorter">
    Show me hotels by star rating
    <select id="rating-sorter">
      <option selected
              value="ascending"
      >
        Low to High (Ascending)
      </option>
      <option value="descending">
        High to Low (Descending)
      </option>
    </select>
  </label>
</div>

`;

exports[`components/RatingSorter should match snapshot for descending 1`] = `

<div class="RatingSorter"
     data-reactroot
>
  <label for="rating-sorter">
    Show me hotels by star rating
    <select id="rating-sorter">
      <option value="ascending">
        Low to High (Ascending)
      </option>
      <option selected
              value="descending"
      >
        High to Low (Descending)
      </option>
    </select>
  </label>
</div>

`;
